component/component.go

```diff
@@ -107,3 +107,25 @@ type ConfigUnmarshaler interface {
 // intoCfg interface{}
 //   An empty interface wrapping a pointer to the config struct to unmarshal into.
 type CustomUnmarshaler func(componentViperSection *viper.Viper, intoCfg interface{}) error
+
+// CreateParamsBase is passed to Create* functions.
+type CreateParamsBase struct {
```

Comment:
Wasn't too sure if I should create this base struct or just repeat the field for all component types as is done for the `Logger` param. Also was considering moving `Logger`, but that would be a breaking change

Reply:
Not sure, if we make this private and embed it in all public (we can also include the Logger), does that work? The fields will be public.

Reply:
Doesn't seem to make a difference if this is private or not:
```
service/builder/exporters_builder.go:327:4: cannot use promoted field createParamsBase.ApplicationStartInfo in struct literal of type component.ExporterCreateParams
service/builder/exporters_builder.go:328:4: cannot use promoted field createParamsBase.Logger in struct literal of type component.ExporterCreateParams
```

Reply:
We can make this work if we change all the related code to not use struct literals, but that would require a breaking change & refactor.

Ok:
```go
creationParams := component.ExporterCreateParams{}
creationParams.ApplicationStartInfo = appInfo
creationParams.Logger = logger
```

Not Ok:
```go
creationParams := component.ExporterCreateParams{
	ApplicationStartInfo: appInfo,
	Logger:               logger,
}
```